dandi/dandiarchive.py

```diff
@@ -363,19 +363,26 @@ class _dandi_url_parser:
         # TODO: Later should better conform to our API, so we could allow
         #       for not only "dandiarchive.org" URLs
         (
-            re.compile(r"DANDI:.*"),
-            {"rewrite": lambda x: "https://identifiers.org/" + x},
-            "DANDI:<dandiset id>",
+            re.compile(
+                fr"DANDI:{DANDISET_ID_REGEX}(?:/[0-9]+\.[0-9]+\.[0-9]+)?",
```

Comment:
Can't `VERSION_REGEX` be used here instead of duplicating regex?

Reply:
@yarikoptic We need to exclude `draft`, which `VERSION_REGEX` includes.

Reply:
I guess I would have made this one to not include it, and ANY_VERSION_REGEX or alike to include it?

Reply:
@yarikoptic Do you want me to do that, or are you concerned about breaking backwards compatibility?

Reply:
to keep compatible, lets go the other way: add `PUBLISHED_VERSION_REGEX = '[0-9]+\.[0-9]+\.[0-9]+'`?

Reply:
@yarikoptic Done.